app/connectors/CodeListConnector.scala

```diff
@@ -73,25 +86,25 @@ class FileBasedCodeListConnector @Inject()(appConfig: AppConfig) extends CodeLis
     (codeItem: CodeItem, locale: Locale) => ProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
-  private val additionalProcedureCodeListsByLang = readCodesFromFile(
+  private val additionalProcedureCodeMapsByLang = readCodesMapFromFile(
     appConfig.additionalProcedureCodes,
     (codeItem: CodeItem, locale: Locale) => AdditionalProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
-  private val additionalProcedureCodeForC21ListsByLang = readCodesFromFile(
+  private val additionalProcedureCodeForC21MapsByLang = readCodesMapFromFile(
     appConfig.additionalProcedureCodesForC21,
     (codeItem: CodeItem, locale: Locale) => AdditionalProcedureCode(codeItem.code, codeItem.getDescriptionByLocale(locale))
   )
 
   def getProcedureCodes(locale: Locale): Seq[ProcedureCode] =
```

Comment:
Personal opinion... :smiley_cat: 
`procedureCodeListsByLang`, `procedureCodeForC21ListsByLang`,  `getProcedureCodes` and `getProcedureCodesForC21` should be placed in `ProcedureCodeService`

Reply:
The CodeListConnector was designed to mimic the type of functionality you would get from a connector that looked up values via a DB or a REST service. The maps are just an efficient way of preventing continually going back to read the json files for each call. I do not see any benefit in creating a service class for each code type, there is no 'business logic' that needs encapsulating in such a class. The whole idea is that in future we could simply replace the `FileBasedCodeListConnector` with an implementation that retrieved the codes from a REST service. If we created these service classes to hold the maps we would have to alter these service classes too (as we probably then would not want to cache the values, or at least not in perpetuity).

Reply:
I do not understand. The `Map`s should be placed in the service classes, not in the connector (irrespective if the loading take place by REST or from the DB). No need to call the connector any more once the codes are retrieved. The service classes will not require any change if we change the connector's implementation, because the service class uses the interface provided by the trait.

Reply:
In my PR I call `CodeListConnector` only once.

Reply:
We should not be discussing what is in your PR as part of this PR.

Reply:
I was using my PR just as an example simply to refute your claim
>  If we created these service classes to hold the maps we would have to alter these service classes too

Reply:
If this connector was not a file based connector (and was instead a DB or REST based connector) then you absolutely would have to call the connector more than once (otherwise what happens if the code list changes!). By caching the values in the service class you are making assumptions about the behavior/implementation of the connector class in the service class (which you should not do). The implementation of the connector class should have no bearing/influence on the service class. The service class should only hold the business logic (like do I need the C21 or standard list).